core/src/utils/content/index.ts

```diff
@@ -0,0 +1,130 @@
+import { componentOnReady } from '../helpers';
+import { printRequiredElementError } from '../logging';
+
+const ION_CONTENT_ELEMENT_SELECTOR = 'ion-content';
+const ION_CONTENT_CLASS_SELECTOR = '.ion-content-scroll-host';
+/**
+ * Selector used for implementations reliant on `<ion-content>` for scroll event changes.
+ *
+ * Developers should use the `.ion-content-scroll-host` selector to target the element emitting
+ * scroll events. With virtual scroll implementations this will be the host element for
+ * the scroll viewport.
+ */
+const ION_CONTENT_SELECTOR = `${ION_CONTENT_ELEMENT_SELECTOR}, ${ION_CONTENT_CLASS_SELECTOR}`;
+
+const hasOverflowScroll = (node: Element | null) => {
```

Comment:
I would feel much more comfortable type casting from: 
```ts
const scrollContainer = getScrollContainer(el as HTMLElement);
```

and removing the type casting elsewhere. Ideally, functions should have narrow types of the explicit things they are dependent on. Otherwise they come highly coupled/harder to use over time. 

Datetime is an example of this, with the `DatetimeParts` type being required on all the utils, even if the util only needs a month to execute.

Reply:
Works for me 👍